cattle/compute/__init__.py

```diff
@@ -63,6 +47,27 @@ def instance_deactivate(self, req=None, instanceHostMap=None,
                                                         progress)
         )
 
+    def get_instance_host_from_map(self, instanceHostMap=None):
+        instance = instanceHostMap.instance
+        host = instanceHostMap.host
+
+        try:
+            host.clusterConnection = \
+                instanceHostMap.data.fields['clusterConnection']
+            log.debug('clusterConnection = %s', host.clusterConnection)
+            host.caCrt = \
+                instanceHostMap.data.fields['caCrt']
+            host.clientCrt = \
+                instanceHostMap.data.fields['clientCrt']
+            host.clientKey = \
+                instanceHostMap.data.fields['clientKey']
+            log.debug('caCrt=%s clientCrt=%s clientKey=%s',
+                      host.caCrt, host.clientCrt, host.clientKey)
+        except (KeyError, AttributeError):
+            pass
```

Comment:
Is there a way on the cattle side to indicate that user wants to enable ssl on the cluster, and pass this info down to the python agent? So if he requests the ssl to be in, we should error out on invalid ssl parameters instead of ignoring them. I am just afraid that by catching all the exceptions and proceeding unauthorized, we might mask the potential bugs + give user an impression that his instance was started with auth turned on, while the agent ignored it.